pkg/process/checks/process_discovery_check_test.go

```diff
@@ -63,6 +90,97 @@ func TestProcessDiscoveryCheck(t *testing.T) {
+		Return(processesByPid, nil)
+
+	expected := []model.MessageBody{
+		&model.CollectorProcDiscovery{
+			ProcessDiscoveries: []*model.ProcessDiscovery{makeProcessDiscoveryModel(t, proc1)},
+			GroupSize:          int32(len(processesByPid)),
+			Host:               &model.Host{},
+		},
+		&model.CollectorProcDiscovery{
+			ProcessDiscoveries: []*model.ProcessDiscovery{makeProcessDiscoveryModel(t, proc2)},
+			GroupSize:          int32(len(processesByPid)),
+			Host:               &model.Host{},
+		},
+		&model.CollectorProcDiscovery{
+			ProcessDiscoveries: []*model.ProcessDiscovery{makeProcessDiscoveryModel(t, proc3)},
+			GroupSize:          int32(len(processesByPid)),
+			Host:               &model.Host{},
+		},
+		&model.CollectorProcDiscovery{
+			ProcessDiscoveries: []*model.ProcessDiscovery{makeProcessDiscoveryModel(t, proc4)},
+			GroupSize:          int32(len(processesByPid)),
+			Host:               &model.Host{},
+		},
+		&model.CollectorProcDiscovery{
+			ProcessDiscoveries: []*model.ProcessDiscovery{makeProcessDiscoveryModel(t, proc5)},
+			GroupSize:          int32(len(processesByPid)),
+			Host:               &model.Host{},
+		},
+	}
+
+	// Test check runs without error
+	actual, err := check.Run(testGroupId(0), &RunOptions{RunStandard: true, NoChunking: false})
+	require.NoError(t, err)
+	assert.ElementsMatch(t, expected, actual.Payloads())
+}
+
+func TestProcessDiscoveryCheckWithoutChunking(t *testing.T) {
+	check, probe := processDiscoveryCheckWithMockProbe(t)
+
+	// Set small chunk size to encourage chunking behavior
+	check.maxBatchSize = 1
+
+	now := time.Now().Unix()
+	proc1 := makeProcessWithCreateTime(1, "git clone google.com", now)
+	proc2 := makeProcessWithCreateTime(2, "mine-bitcoins -all -x", now+1)
+	proc3 := makeProcessWithCreateTime(3, "foo --version", now+2)
+	proc4 := makeProcessWithCreateTime(4, "foo -bar -bim", now+3)
+	proc5 := makeProcessWithCreateTime(5, "datadog-process-agent --cfgpath datadog.conf", now+2)
+
+	processesByPid := map[int32]*procutil.Process{1: proc1, 2: proc2, 3: proc3, 4: proc4, 5: proc5}
+	probe.On("ProcessesByPID", mock.Anything, mock.Anything).
+		Return(processesByPid, nil)
+
+	expected := []*model.ProcessDiscovery{
+		makeProcessDiscoveryModel(t, proc1),
+		makeProcessDiscoveryModel(t, proc2),
+		makeProcessDiscoveryModel(t, proc3),
+		makeProcessDiscoveryModel(t, proc4),
+		makeProcessDiscoveryModel(t, proc5),
+	}
+
+	// Test check runs without error
+	actual, err := check.Run(testGroupId(0), &RunOptions{RunStandard: true, NoChunking: true})
+	require.NoError(t, err)
+
+	// Assert to check there is only one chunk and that the nested values of this chunk match expected
+	assert.Len(t, actual.Payloads(), 1)
+	actualPayloads := actual.Payloads()
+	assert.IsType(t, &model.CollectorProcDiscovery{}, actualPayloads[0])
+	collectorProcDiscovery := actualPayloads[0].(*model.CollectorProcDiscovery)
+	ProcessDiscoveries := collectorProcDiscovery.GetProcessDiscoveries()
```

Comment:
same comment regarding trying to reduce this into a table test if possible. `ProcessDiscoveries` should be lowercased `processDiscoveries`, but try to make this a table test first.